.github/workflows/doc-build.yml

```diff
@@ -99,19 +91,29 @@ jobs:
         REF_TYPE=${{ github.ref_type }}
         REF_NAME=${{ github.ref_name }}
 
+        # If it's main branch, add noindex tag to all .html files to exclude from Google Search indexing.
+        REF_NAME=$(echo "${{ github.ref }}")
+        echo "Ref name: ${REF_NAME}"
+        if [[ "${{ github.ref }}" == 'refs/heads/main' ]]; then
+          find docs -name "*.html" -print0 | xargs -0 sed -i '/<head>/a \ \ <meta name="robots" content="noindex">';
+        fi
+
         # If building for a release tag, branch, set the branch/tag name
         # as the target folder in the gh-pages branch. The artifacts created
         # during the build will be copied over to the target dir in the
         # gh-pages branch.
         if [[ "${REF_TYPE}" == branch ]]; then
           TARGET_FOLDER="${REF_NAME}"
         elif [[ "${REF_TYPE}" == tag ]]; then
-          # Strip the leading "v" as well as the trailing patch version and "-rc" suffix.
-          # For example: 'v0.1.2' -> '0.1' and 'v0.1.2-rc1' -> 0.1.
-          TARGET_FOLDER=$(echo "${REF_NAME}" | sed 's/^v//i; s/-rc[0-9]*$//; s/\.[0-9]*$//')
-        else
-          echo "ERROR: Invalid REF_TYPE: ${REF_TYPE}. Expected 'branch' or 'tag'."
-          exit 1
+          case "${REF_NAME}" in
+            *-rc*)
+              echo "Aborting upload since this is an RC tag: ${REF_NAME}"
```

Comment:
Please add a comment explaining why it makes sense to abort when encountering a tag. Is it because we will already generate the docs when the release branch is pushed?

Reply:
yes, I believe this is what we do across other repositories - no documentation will be generated for v1.0.0-rc1 or similar.